lib/api/apiUtils/object/createAndStoreObject.js

```diff
@@ -196,13 +214,11 @@ function createAndStoreObject(bucketName, bucketMD, objectKey, objMD, authInfo,
                 dataGetInfoArr[0].cipheredDataKey =
                     cipherBundle.cipheredDataKey;
             }
-            if (dataStoreSize !== undefined) {
```

Comment:
These conditions were added specifically for the `mdonly` header use originally but are consolidated with these code changes.
https://github.com/scality/cloudserver/pull/1609/files#diff-e45defa8c8c53634533cf165cc9232ceR196